File: python/zigZagConversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if not s:
            return ''
        if numRows==1 or len(s)<numRows:
            return s

        zigzag_list = ['' for row in range(numRows)]
        row = 0
        flag = True
        zigzag = ''
        
        for char in s:
            zigzag_list[row] += char
            if flag:
                row += 1
            else:
                row -= 1
            if row==numRows:
                flag = False
                row = numRows-2
            if row==-1:
                flag = True
                row = 1
                
        for row in zigzag_list:
            zigzag += row
        
        return zigzag

File: java/zigZagConversion.java
class Solution {
    public String convert(String s, int numRows) {
        String zigzag = "";
        String[] zigzag_list = new String[numRows];
        ArrayList<String> l = new ArrayList<>(3);
        boolean flag = true;
        int row = 0;
        
        if(s=="")
            return "";
        if(s.length()<numRows || numRows==1) 
            return s;
        
        Arrays.fill(zigzag_list, "");
        
        for(int i=0; i<s.length(); i++) {
            zigzag_list[row] += s.charAt(i);
            if(flag) {
                row++;
            } else {
                row--;
            }
            if(row==numRows) {
                flag = false;
                row = numRows-2;
            } 
            if(row==-1) {
                flag = true;
                row = 1;
            }
        }

        for(int i=0;i<numRows; i++) 
            zigzag += zigzag_list[i];
        
        return zigzag;
        
    }
}